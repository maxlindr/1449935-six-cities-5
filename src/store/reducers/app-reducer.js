import {ActionType} from '../action';

const initialState = {
  alertMessage: null,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SHOW_ALERT:
      return Object.assign({}, state, {
        alertMessage: action.payload,
      });

    case ActionType.CLOSE_ALERT:
      return Object.assign({}, state, {
        alertMessage: null,
      });
  }

  return state;
};
