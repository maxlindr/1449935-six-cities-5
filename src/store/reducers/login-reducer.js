import {ActionType} from '../action';

const initialState = {
  isLoginFailedWithUnauthorized: false,
};

export default (state = initialState, action) => {
  if (action.type === ActionType.SET_LOGIN_FAILED) {
    return Object.assign({}, state, {
      isLoginFailedWithUnauthorized: action.payload
    });
  }

  return state;
};
