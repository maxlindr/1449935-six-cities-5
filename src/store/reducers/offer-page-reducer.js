import {ActionType} from '../action';

const initialState = {
  reviews: null,
  nearbyPlaces: null,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_FETCHED_REVIEWS:
      return Object.assign({}, state, {
        reviews: action.payload
      });

    case ActionType.SET_FETCHED_NEARBY_PLACES:
      return Object.assign({}, state, {
        nearbyPlaces: action.payload
      });
  }

  return state;
};
