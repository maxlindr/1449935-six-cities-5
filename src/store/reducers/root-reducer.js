import {combineReducers} from 'redux';
import dataReducer from './data-reducer/data-reducer';
import userReducer from './user-reducer/user-reducer';
import offerPageReducer from './offer-page-reducer/offer-page-reducer';
import loginReducer from './login-reducer/login-reducer';
import errorPageReducer from './error-page-reducer/error-page-reducer';
import appReducer from './app-reducer/app-reducer';

export const StateNameSpace = {
  DATA: `DATA`,
  USER: `USER`,
  OFFER_PAGE: `OFFER_PAGE`,
  LOGIN_PROCESSING: `LOGIN_PROCESSING`,
  ERROR_PAGE: `ERROR_PAGE`,
  APP: `APP`
};

export default combineReducers({
  [StateNameSpace.DATA]: dataReducer,
  [StateNameSpace.USER]: userReducer,
  [StateNameSpace.OFFER_PAGE]: offerPageReducer,
  [StateNameSpace.LOGIN_PROCESSING]: loginReducer,
  [StateNameSpace.ERROR_PAGE]: errorPageReducer,
  [StateNameSpace.APP]: appReducer,
});
