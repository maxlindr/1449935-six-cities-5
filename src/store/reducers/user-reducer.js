import {AuthorizationStatus} from '../../constants';
import {ActionType} from '../actions/action';

const initialState = {
  user: null,
  authorizationStatus: AuthorizationStatus.NOT_AUTHORIZED,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_AUTHORIZATION_STATUS:
      return Object.assign({}, state, {
        authorizationStatus: action.payload,
      });

    case ActionType.SET_USER_DATA:
      return Object.assign({}, state, {
        user: action.payload,
      });
  }

  return state;
};

export default reducer;
