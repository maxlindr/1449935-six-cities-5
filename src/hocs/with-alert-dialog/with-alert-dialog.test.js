import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import withAlertDialogConnected, {withAlertDialog} from './with-alert-dialog';
import {StateNameSpace} from '../../store/reducers/root-reducer';

const mockStore = configureStore([]);
let store;

const MockComponent = () => <div>Mock component</div>;
const MockComponentWrapped = withAlertDialog(MockComponent);
const MockComponentWrappedConnected = withAlertDialogConnected(MockComponent);

describe(`withAlertDialog`, () => {
  const MESSAGE_PROP_NAME = `alertDialogMessage`;

  describe(`Не подключенный к store`, () => {
    it(`не должен добавлять компонент Alert при отсутствии пропа ${MESSAGE_PROP_NAME}`, () => {
      const tree = renderer.create((
        <MockComponentWrapped closeAlertDialog={() => {}}/>
      )).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`должен добавлять компонент Alert при наличии пропа ${MESSAGE_PROP_NAME}`, () => {
      const tree = renderer.create((
        <MockComponentWrapped
          alertDialogMessage={`Message`}
          closeAlertDialog={() => {}}
        />
      )).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe(`Подключенный к store`, () => {
    const MESSAGE_STORE_KEYS_CHAIN = `store.${StateNameSpace.APP}.alertMessage`;

    it(`не должен добавлять компонент Alert при отсутствии данных в ${MESSAGE_STORE_KEYS_CHAIN}`, () => {
      store = mockStore({
        [StateNameSpace.APP]: {
          alertMessage: null,
        }
      });

      const tree = renderer.create((
        <Provider store={store}>
          <MockComponentWrappedConnected />
        </Provider>
      )).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`должен добавлять компонент Alert при наличии данных в ${MESSAGE_STORE_KEYS_CHAIN}`, () => {
      store = mockStore({
        [StateNameSpace.APP]: {
          alertMessage: `Message`
        }
      });

      const tree = renderer.create((
        <Provider store={store}>
          <MockComponentWrappedConnected />
        </Provider>
      )).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
