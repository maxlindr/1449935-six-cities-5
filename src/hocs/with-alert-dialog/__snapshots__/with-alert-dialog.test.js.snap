// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAlertDialog Не подключенный к store должен добавлять компонент Alert при наличии пропа alertDialogMessage 1`] = `
Array [
  <div>
    Mock component
  </div>,
  <aside
    className="alert"
  >
    <div
      className="alert__window"
    >
      <div
        className="alert__container"
      >
        <p
          className="alert__message"
        >
          Message
        </p>
      </div>
      <button
        className="button alert__button"
        onClick={[Function]}
        type="button"
      >
        OK
      </button>
    </div>
  </aside>,
]
`;

exports[`withAlertDialog Не подключенный к store не должен добавлять компонент Alert при отсутствии пропа alertDialogMessage 1`] = `
<div>
  Mock component
</div>
`;

exports[`withAlertDialog Подключенный к store должен добавлять компонент Alert при наличии данных в store.APP.alertMessage 1`] = `
Array [
  <div>
    Mock component
  </div>,
  <aside
    className="alert"
  >
    <div
      className="alert__window"
    >
      <div
        className="alert__container"
      >
        <p
          className="alert__message"
        >
          Message
        </p>
      </div>
      <button
        className="button alert__button"
        onClick={[Function]}
        type="button"
      >
        OK
      </button>
    </div>
  </aside>,
]
`;

exports[`withAlertDialog Подключенный к store не должен добавлять компонент Alert при отсутствии данных в store.APP.alertMessage 1`] = `
<div>
  Mock component
</div>
`;
